openslides_backend/action/actions/speaker/delete.py

```diff
@@ -34,3 +40,30 @@ def check_permissions(self, instance: dict[str, Any]) -> None:
             if meeting_user.get("user_id") == self.user_id and not len(restricted):
                 return
         super().check_permissions(instance)
+
+    def get_updated_instances(self, action_data: ActionData) -> ActionData:
+        self.end_time = round(time())
+        return super().get_updated_instances(action_data)
+
+    def update_instance(self, instance: dict[str, Any]) -> dict[str, Any]:
+        speaker = self.datastore.get(
+            fqid_from_collection_and_id("speaker", instance["id"]),
+            [
+                "meeting_id",
+                "begin_time",
+                "end_time",
+                "pause_time",
+                "speech_state",
+                "point_of_order",
+                "unpause_time",
+                "structure_level_list_of_speakers_id",
+            ],
+        )
+        if (
+            speaker.get("begin_time")
+            and not speaker.get("end_time")
+            and not speaker.get("pause_time")
+        ):
+            self.decrease_structure_level_countdown(self.end_time, speaker)
```

Comment:
if the end_time is only looked up here, then you should get it here and use round(time()) as function call parameter. As consequence the changes in lines 43-47 are not necessary.

Reply:
Then there may be discrepancies when multiple speakers are stopped at once

Reply:
More than 1 speaker talking at the same would be no good idea at all;-)